### DISCUSSION TITLE
Unable to read/flash Vue - Pulling my hair out

### DISCUSSION THREAD

**Discussion by KidA001** [emporia-vue-local/esphome#223] [Q&A] [answered]
I am pretty experienced with ESPs, and in fact this is my second Vue 2 device. I flashed my first one a couple days ago and it went mostly smooth.

I am unable to backup or flash my Vue2 with my USB to serial adapter (same one I've been using). I have verified that there is continuity between the pins on the board and my USB Serial device. I even replaced the cables just to make sure. I have followed [this thread](https://github.com/emporia-vue-local/esphome/discussions/31) to bypass the resistor on the TX pin which I had to do on my last vue. I know IO0 is pulled to ground.

No matter what I do it cannot read from the ESP on the board, but it is powered up. Running `esptool.py flash_id` returns nothing. When I'm watching the RX TX pins on my UART device it shows that the ESP is never sending anything out, though it is transmitting to the board.

Any help appreciated

Is there something else I'm missing?

**Comment by flaviut** [comment]
IIRC certain boards have series resistors that don't work well with certain serial dongles. Perhaps you can try soldering directly to the esp32 rx/tx pins.

**Reply by KidA001** [reply]
@flaviut thanks for the reply, just soldered directly (man those are small pins!) but still no dice. Verified the continuitiy between my soldered wires and the original header pins and they are connected. also made sure I didn't short them to any other pins as well.

So I'm guessing either this device is bricked or my UART devices is broken.

**Reply by clowrey** [reply]
Have you tried applying 5V to the Vue2 from another source? Some PC USB ports cannot support enough current for the device.

**Reply by KidA001** [reply]
I had used it previously on another Vue without issue, but yes I can try an external source for 5V and see if that makes a difference. Appreciate the help

**Reply by KidA001** [reply]
External 5V source did not make a difference

**Reply by flaviut** [reply]
yeah, I'm sorry :(

IDK what's going on here. Maybe the next step in debugging this (if you'd like) is powering the vue off an external power supply (the inrush power-on current always resets my serial dongle) while being connected via UART. The bootloader I believe prints some details about the boot process.

But I certainly don't have a well-tested procedure here or any really concrete ideas.

**Reply by KidA001** [reply]
I was wondering how to get logging from the device, but not totally sure. Honestly I have a new one coming tomorrow that I will try it on and swap it out of it works on the new one. If it doesn't I'll see about replacing my UART device. Appreciate the help, i'll post back what I find.

**Reply by KidA001** [reply, marked as answer]
Just a follow up - I received a new unit today and after soldering headers on and holding a pin by hand to the Tx pin on the board, I was able to flash it first try. I'll be sending the other unit back. Appreciate the help here

**Reply by clowrey** [reply]
Personally I wouldn't send a unit back that I likely broke myself... Replace the ESP32 module.

**Reply by KidA001** [reply]
Unit works fine on it's own when powered and can pair with the Emporia Vue app, it just can be custom flashed with custom firware for reasons that are unclear to me.
